library/src/main/java/com/liulishuo/filedownloader/services/BaseFileServiceUIGuard.java

```diff
@@ -157,7 +158,10 @@ public void bindStartByContext(final Context context, final Runnable connectedRu
             bindContexts.add(context);
         }
 
-        runServiceForeground = FileDownloadUtils.needMakeServiceForeground(context);
+        boolean needMakeServiceForeground = FileDownloadUtils.needMakeServiceForeground(context);
+        boolean haveWaitingTask =
+                FileDownloader.getImpl().getLostConnectedHandler().haveWaitingTask();
+        runServiceForeground = needMakeServiceForeground && haveWaitingTask;
```

Comment:
`needMakeServiceForeground && haveWaitingTask` means not only need FileDownloadUtils configured but also meet has waiting task, which not the same of the title of MR said.

Reply:
The title just description the problem, I will change it.

Reply:
Updated!